deploy/stacks/cloudfront.py

```diff
@@ -365,7 +353,7 @@ def __init__(
                     route53_targets.CloudFrontTarget(cloudfront_distribution)
                 ),
             )
-        if userguide_alternate_domain:
+        if userguide_alternate_domain and custom_auth != None:
```

Comment:
As a suggestion, maybe we can simplify the way we are executing the custom_auth condition. At the moment we are evaluating the condition in multiple places, maybe we can group together the code that is relevant for the userguide. Something like:

```python
        if frontend_alternate_domain:
            frontend_record = route53.ARecord(
                self,
                'CloudFrontFrontendDomain',
                record_name=frontend_alternate_domain,
                zone=hosted_zone,
                target=route53.RecordTarget.from_alias(
                    route53_targets.CloudFrontTarget(cloudfront_distribution)
                ),
            )
        user_docs_bucket = None
        if custom_auth is None:
            userguide_docs_distribution, user_docs_bucket = self.build_static_site(
                f'userguide',
                acl,
                auth_at_edge,
                envname,
                resource_prefix,
                userguide_domain_names,
                certificate,
                ssl_support_method,
                security_policy,
                logging_bucket,
                custom_auth
            )
            self.user_docs_bucket = user_docs_bucket
            self.user_docs_distribution = userguide_docs_distribution

            if userguide_alternate_domain:
                userguide_record = route53.ARecord(
                    self,
                    'CloudFrontUserguideDomain',
                    record_name=userguide_alternate_domain,
                    zone=hosted_zone,
                    target=route53.RecordTarget.from_alias(
                        route53_targets.CloudFrontTarget(userguide_docs_distribution)
                    ),
                )

        if tooling_account_id:
            cross_account_deployment_role = iam.Role(
                self,
                f'S3DeploymentRole{envname}',
                role_name=f'{resource_prefix}-{envname}-S3DeploymentRole',
                assumed_by=iam.AccountPrincipal(tooling_account_id),
            )
            resources_for_cross_account = []
            resources_for_cross_account.append(f'{cloudfront_bucket.bucket_arn}/*')
            if user_docs_bucket:
                resources_for_cross_account.append(f'{user_docs_bucket.bucket_arn}/*')

```

Reply:
Addressed this by clubbing changes together which are used for creating user guide